src/Platforms/AbstractPlatform.php

```diff
@@ -802,6 +804,11 @@ public function createUnionSQLBuilder(): UnionSQLBuilder
         return new DefaultUnionSQLBuilder($this);
     }
 
+    public function createWithSQLBuilder(): WithSQLBuilder
+    {
+        return new DefaultWithSQLBuilder();
+    }
```

Comment:
If we want to give MySQL 5.7 users a nice exception, we could override this method in `MySQLPlatform`:

```php
public function createWithSQLBuilder(): WithSQLBuilder
{
    throw NotSupported::new(__METHOD__);
}
```

… and override it again in `MySQL80Platform`:

```php
public function createWithSQLBuilder(): WithSQLBuilder
{
    return AbstractMySQLPlatform::createWithSQLBuilder();
}
```

Reply:
I don't know. What do you think of that @morozov @sbuerk ?

Reply:
Sounds good. We do that in other places as well. This is a bit odd but this is due to platforms inheriting one another.